lib/sync-service/client.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * NOTE: This is only for backward compatibility with the existing demo apps.
+ *
+ * When this functionality has been moved to modules, then this can be removed.
+ *
+ * @type {_|exports|module.exports}
+ * @private
+ */
+
+var q = require('q');
+var _ = require('lodash');
+var shortid = require('shortid');
+var CONSTANTS = require('../constants');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+var mediator, manager, workorderSyncSubscribers;
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {*}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
```

Comment:
Probably a minor nitpick, but in Node.js everything is Error, Done.  Reversing it here seems like it will cause a bunch of errors in the future.